src/SettingsServiceProvider.php

```diff
@@ -32,25 +32,39 @@ class SettingsServiceProvider extends ServiceProvider
      */
     public function boot()
     {
+        // use the vendor configuration file as fallback
+        $this->mergeConfigFrom(
+            __DIR__.'/config/backpack/settings.php',
+            'backpack.settings'
+        );
+
         // define the routes for the application
         $this->setupRoutes($this->app->router);
 
         // only use the Settings package if the Settings table is present in the database
-        if (!\App::runningInConsole() && count(Schema::getColumnListing('settings'))) {
+        if (!\App::runningInConsole() && count(Schema::getColumnListing(config('backpack.settings.table_name')))) {
             // get all settings from the database
             $settings = Setting::all();
 
+            $config_prefix = config('backpack.settings.config_prefix');
+
             // bind all settings to the Laravel config, so you can call them like
             // Config::get('settings.contact_email')
             foreach ($settings as $key => $setting) {
-                Config::set('settings.'.$setting->key, $setting->value);
+                $prefixed_key = !empty($config_prefix) ? $config_prefix.'.'.$setting->key : $setting->key;
+                Config::set($prefixed_key, $setting->value);
             }
         }
         // publish the migrations and seeds
-        $this->publishes([__DIR__.'/database/migrations/' => database_path('migrations')], 'migrations');
+        $this->publishes([
+            __DIR__.'/database/migrations/create_settings_table.php.stub' => database_path('migrations/'.date('Y_m_d_His', time()).'_create_settings_table.php'),
+        ], 'migrations');
```

Comment:
@pxpm if the dev publish this package multiple times will this create multiple migrations?
And if so, should it?

Previously it would replace the existing file, that was the same (same name and content) so that was not an issue.

Reply:
It would, and it should NOT!

I've just added one more configuration option for migration name and I think we are done here.

Thanks for the review, I didn't noticed it previously.